FirebaseMLModelDownloader/Sources/ModelDownloadTask.swift

```diff
@@ -0,0 +1,163 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import FirebaseCore
+
+enum DownloadStatus {
+  case notStarted
+  case inProgress
+  case completed
+}
+
+/// Progress and completion handlers for a model download.
+class DownloadHandlers {
+  typealias ProgressHandler = (Float) -> Void
+  typealias Completion = (Result<CustomModel, DownloadError>) -> Void
+
+  var progressHandler: ProgressHandler?
+  var completion: Completion
+
+  init(progressHandler: ProgressHandler?, completion: @escaping Completion) {
+    self.progressHandler = progressHandler
+    self.completion = completion
+  }
+}
+
+/// Manager for model downloads.
```

Comment:
A short description of the class responsibilities may be helpful here. E.g. the class is responsible for downloading the model file and updating the model persistent storage.

Reply:
Done.